server/coordinator/procedure/storage_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package procedure
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/CeresDB/ceresmeta/server/etcdutil"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	TestClusterID       = 1
+	TestRootPath        = "/rootPath"
+	DefaultTimeout      = time.Second * 10
+	DefaultScanBatchSie = 100
+)
+
+func newTestStorage(t *testing.T) *EtcdStorageImpl {
+	_, client, _ := etcdutil.PrepareEtcdServerAndClient(t)
+	storage := NewEtcdStorageImpl(client, uint32(TestClusterID), TestRootPath)
+	return storage
+}
+
+func testWrite(t *testing.T, storage *EtcdStorageImpl) {
+	re := require.New(t)
+	ctx, cancel := context.WithTimeout(context.Background(), DefaultTimeout)
+	defer cancel()
+
+	testMeta1 := &Meta{
+		ID:      uint64(1),
+		Typ:     TransferLeader,
+		State:   StateInit,
+		RawData: []byte("test"),
+	}
+
+	// Test create new procedure
+	err := storage.CreateOrUpdate(ctx, testMeta1)
+	re.NoError(err)
+
+	testMeta2 := &Meta{
+		ID:      uint64(2),
+		Typ:     TransferLeader,
+		State:   StateInit,
+		RawData: []byte("test"),
+	}
+	err = storage.CreateOrUpdate(ctx, testMeta2)
+	re.NoError(err)
+
+	// Test update procedure
+	testMeta2.RawData = []byte("test update")
+	err = storage.CreateOrUpdate(ctx, testMeta2)
+	re.NoError(err)
+}
+
+func testScan(t *testing.T, storage *EtcdStorageImpl) {
+	re := require.New(t)
+	ctx, cancel := context.WithTimeout(context.Background(), DefaultTimeout)
+	defer cancel()
+
+	metas := make([]*Meta, 0)
+	err := storage.ReadAll(ctx, DefaultScanBatchSie, &metas)
+	re.NoError(err)
+	re.Equal(2, len(metas))
```

Comment:
Maybe it is not enough to do only length check, and shall do a more content check on the `metas`?